[Bug]: Resolver fails when the existing requirements.txt does not end in a newline character

### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

`openhands-resolver.yml` fails on te Install "OpenHands" job when there already exists a `requirements.txt` file in the root level directory of the repository with the following error:
```
Run if [[ "fix-me" == "fix-me-experimental" ]] ||
ERROR: Invalid requirement: 'openai==1.1.0openhands-ai==0.15.2': Expected end or semicolon (after version specifier)
    openai==1.1.0openhands-ai==0.15.2
          ~~~~~~~^ (from line 40 of requirements.txt)
Error: Process completed with exit code 1.
```
where openai happpens to be our requirements.txt's last dependency:
```
aiohttp==3.8.4
aiosignal==1.3.1
async-timeout==4.0.2
attrs==23.1.0
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
cryptography==43.0.1
dataclasses-json==0.5.8
Deprecated==1.2.14
frozenlist==1.3.3
greenlet==2.0.2
idna==3.4
langchain==0.3.7
langchainplus-sdk==0.0.17
marshmallow==3.19.0
marshmallow-enum==1.5.1
multidict==6.0.4
mypy-extensions==1.0.0
numexpr==2.8.5
numpy==1.24.3
openapi-schema-pydantic==1.2.4
packaging==23.1
pycparser==2.21
pydantic==1.10.9
PyGithub==1.58.2
PyJWT==2.7.0
pylint==2.17.4
PyNaCl==1.5.0
PyYAML==6.0
requests==2.31.0
SQLAlchemy==2.0.16
tenacity==8.2.2
tqdm==4.65.0
typing-inspect==0.9.0
typing_extensions==4.6.3
urllib3==2.0.3
wrapt==1.15.0
yarl==1.9.2
openai==1.1.0
```

The erroneous code is actually in
```
      - name: Get latest versions and create requirements.txt
        run: |
          python -m pip index versions openhands-ai > openhands_versions.txt
          OPENHANDS_VERSION=$(head -n 1 openhands_versions.txt | awk '{print $2}' | tr -d '()')
          echo "openhands-ai==${OPENHANDS_VERSION}" >> requirements.txt
          cat requirements.txt
```
which will result in the last line, in our example, being:
```
openai==1.1.0openhands-ai==0.15.2
```

### OpenHands Installation

Development workflow

### OpenHands Version

0.15.2

### Operating System

Linux

### Logs, Errors, Screenshots, and Additional Context

`openhands-resolver.yml` fails on te Install "OpenHands" job when there already exists a `requirements.txt` file in the root level directory of the repository with the following error:
```
Run if [[ "fix-me" == "fix-me-experimental" ]] ||
ERROR: Invalid requirement: 'openai==1.1.0openhands-ai==0.15.2': Expected end or semicolon (after version specifier)
    openai==1.1.0openhands-ai==0.15.2
          ~~~~~~~^ (from line 40 of requirements.txt)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/openhands-resolver.yml
+++ b/.github/workflows/openhands-resolver.yml
@@ -84,6 +84,10 @@ jobs:
         run: |
           python -m pip index versions openhands-ai > openhands_versions.txt
           OPENHANDS_VERSION=$(head -n 1 openhands_versions.txt | awk '{print $2}' | tr -d '()')
+          # Ensure requirements.txt ends with newline before appending
+          if [ -f requirements.txt ] && [ -s requirements.txt ]; then
+            sed -i -e '$a\' requirements.txt
+          fi
           echo "openhands-ai==${OPENHANDS_VERSION}" >> requirements.txt
           cat requirements.txt
 

